data-prepper-core/src/integrationTest/java/org/opensearch/dataprepper/integration/PipelinesWithAcksIT.java

```diff
@@ -130,7 +131,7 @@ void three_pipelines_with_route_and_multiple_records() {
                 .untilAsserted(() -> {
             List<Record<Event>> outputRecords = inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER);
             assertThat(outputRecords, not(empty()));
-            assertThat(outputRecords.size(), equalTo(numRecords));
+            assertThat(outputRecords.size(), lessThanOrEqualTo(numRecords));
```

Comment:
I'm curious about why you made this change? Is this too lenient?

Reply:
No. Some records will not be sent to sink. But I will create a new test for it and keep the existing test untouched.